Purchases/Purchasing/RCProductInfoExtractor.m

```diff
@@ -0,0 +1,148 @@
+//
+// Created by Andrés Boedo on 5/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+#import "RCProductInfoExtractor.h"
+#import "RCProductInfo.h"
+#import "RCISOPeriodFormatter.h"
+#import "RCPromotionalOffer.h"
+#import "NSLocale+RCExtensions.h"
+#import <StoreKit/StoreKit.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+
+@interface RCProductInfoExtractor ()
+
+@property (nonatomic) RCISOPeriodFormatter *formatter API_AVAILABLE(ios(11.2), macos(10.13.2), tvos(11.2));
+
+@end
+
+
+@implementation RCProductInfoExtractor
+
+# pragma mark - public methods
+
+- (instancetype)init {
+    if (self = [super init]) {
+        if (@available(iOS 11.2, macOS 10.13.2, tvOS 11.2, *)) {
+            self.formatter = [[RCISOPeriodFormatter alloc] init];
+        }
+    }
+    return self;
+}
+
+- (RCProductInfo *)extractInfoFromProduct:(SKProduct *)product {
```

Comment:
most of this file was extracted from RCPurchases and cleaned up so that it's easier to follow